### ISSUE TITLE
io.netty.util.Version#identify not working in uber-jars

### ISSUE THREAD

**Issue by sebthom** [quarkusio/quarkus#16662]
## Describe the bug
When packaging a Quarkus application as an [uber-jar](https://quarkus.io/guides/maven-tooling#uber-jar-maven) the [io.netty.util.Version#identify](https://netty.io/4.1/api/io/netty/util/Version.html#identify--) method returns an empty map instead of the versions of the included netty artifacts. The reason is that each Netty artifact contains a `META-INF/io.netty.versions` which is not included in the uber-jar.

`io.netty.util.Version#identify` not reporting the actual Netty version for example prevents the app from being monitored via Dynatrace.

### Expected behavior
The fat-jar includes a `META-INF/io.netty.versions` file which contains the merged content of all `META-INF/io.netty.versions` of the included netty artifacts to ensure the correct behaviour of the method [io.netty.util.Version#identify](https://netty.io/4.1/api/io/netty/util/Version.html#identify--).

### Actual behavior
When programmatically executing [io.netty.util.Version#identify](https://netty.io/4.1/api/io/netty/util/Version.html#identify--) an empty map is returned instead of the version.

We also see the following Dynatrace error in the logs:
```
2021-04-12 14:13:41.565 UTC [57bf881a] warning [java  ] [agent     ] Failed to detect Netty version
2021-04-12 14:13:41.649 UTC [57bf881a] warning [java  ] [netty     ] Caught exception in IntrospectionBase
 java.lang.NullPointerException
 	at com.compuware.apm.agent.introspection.netty.NettyIntrospection$1.run(NettyIntrospection.java:167)
 	at com.compuware.apm.agent.introspection.IntrospectionBase.introspect(IntrospectionBase.java:300)
 	at com.compuware.apm.agent.introspection.IntrospectionBase.introspect(IntrospectionBase.java:213)
 	at com.compuware.apm.agent.introspection.netty.NettyIntrospection.HttpRequestReceivedEnter(NettyIntrospection.java:158)
 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java)
 	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
 	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:61)
 	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
 	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
 	at java.base/java.lang.Thread.run(Unknown Source)
```

## To Reproduce

1. download and unzip this sample application [netty-version-uberjar.zip](https://github.com/quarkusio/quarkus/files/6346243/netty-version-uberjar.zip)
2. run `mvn compile quarkus:dev` which will print something like:
```
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2021-04-20 21:27:41,146 INFO  [io.quarkus] (Quarkus Main Thread) netty-version-uberjar 1.0.0-SNAPSHOT on JVM (powered by Quarkus 1.13.2.Final) started in 4.023s. Listening on: http://localhost:8080
2021-04-20 21:27:41,166 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-04-20 21:27:41,168 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi]
************************************
Netty Versions: {netty-buffer=netty-buffer-4.1.49.Final.d0ec961, netty-codec=netty-codec-4.1.49.Final.d0ec961, netty-codec-dns=netty-codec-dns-4.1.49.Final.d0ec961, netty-codec-http=netty-codec-http-4.1.49.Final.d0ec961, netty-codec-http2=netty-codec-http2-4.1.49.Final.d0ec961, netty-codec-socks=netty-codec-socks-4.1.49.Final.d0ec961, netty-common=netty-common-4.1.49.Final.d0ec961, netty-handler=netty-handler-4.1.49.Final.d0ec961, netty-handler-proxy=netty-handler-proxy-4.1.49.Final.d0ec961, netty-resolver=netty-resolver-4.1.49.Final.d0ec961, netty-resolver-dns=netty-resolver-dns-4.1.49.Final.d0ec961, netty-transport=netty-transport-4.1.49.Final.d0ec961}
************************************
```
3. create the uberjar using `mvn package`
4. run `java -jar target\netty-version-uberjar-1.0.0-SNAPSHOT-runner.jar` which will print something like:
```
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2021-04-20 21:29:04,225 INFO  [io.quarkus] (main) netty-version-uberjar 1.0.0-SNAPSHOT on JVM (powered by Quarkus 1.13.2.Final) started in 2.060s. Listening on: http://0.0.0.0:8080
2021-04-20 21:29:04,257 INFO  [io.quarkus] (main) Profile prod activated.
2021-04-20 21:29:04,259 INFO  [io.quarkus] (main) Installed features: [cdi]
************************************
Netty Versions: {}
************************************
2021-04-20 21:29:04,298 INFO  [io.quarkus] (main) netty-version-uberjar stopped in 0.033s
```

### Output of `uname -a` or `ver`
Linux app-31-rq52p 3.10.0-1160.6.1.el7.x86_64 #1 SMP Wed Oct 21 13:44:38 EDT 2020 x86_64 GNU/Linux

### Output of `java -version`
openjdk version "11.0.10" 2021-01-19 LTS

### Quarkus version or git rev
Quarkus 1.13.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)

**Comment by gsmet** [comment]
https://github.com/quarkusio/quarkus/pull/16664 should fix it, thanks for the report.

**Comment by sebthom** [comment]
@gsmet thanks for the quick fix! do you think this can make it into 1.13.3 or 1.14?

**Comment by gsmet** [comment]
Yes, it will be backported.
